src/mam4xx/utils.hpp

```diff
@@ -209,7 +217,9 @@ void extract_stateq_from_prognostics(const mam4::Prognostics &progs,
   for (int m = 0; m < AeroConfig::num_modes(); ++m) {
     // First add the aerosol species mmr
     for (int a = 0; a < mam4::num_species_mode(m); ++a) {
-      q[s_idx] = progs.q_aero_i[m][a](klev);
+      const int h_ind = get_haero_prognostics_index(
+          m, a); // Index of HAERO's prognostics array
+      q[s_idx] = progs.q_aero_i[m][h_ind](klev);
```

Comment:
The code will invoke `get_haero_prognostics_index` inside of levels (kk) and cols (icols) loop. We must check if this part of code becomes a bottleneck.

Reply:
Is there an easy way to check it? To me, it looked more like an array look-up. Do you think it might cause noticeable slowdowns?

Reply:
I hope that @cameronrutherford will tells if this type of code causes slowdowns. For my understanding, he will carry out a performance analysis for mam4xx.